clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java

```diff
@@ -301,6 +301,24 @@ public void testInvalidKeystorePassword() throws Exception {
             // Expected exception
         }
     }
+
+    /**
+     * Tests that client connections can be created to a server
+     * if null truststore password is used
+     */
+    @Test
+    public void testNullTruststorePassword() throws Exception {
```

Comment:
I see, thanks for the explanation. OK, so this means that if someone mistypes the key for the truststore password, integrity checking is disabled and the unused config warning is the only clue to the user. Probably OK.